### PR TITLE
ci(release): use new set of credentials for GPG and Maven Central

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Use the Vault secrets accessed to run the Release/Snapshots.

Those secrets are managed by the Release team and they provided specific ones for this project.

## Checklist

- [ ] This is an enhancement of existing features, or a new feature in existing plugins
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] I have made corresponding changes to the documentation
- [ ] This is a bugfix
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that would fail without this fix
- [ ] This is a new plugin
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] My code follows the [style guidelines of this project](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#java-language-formatting-guidelines)
  - [ ] I have made corresponding changes to the documentation
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#testing) pass locally with my changes
  - [ ] I have updated [supported-technologies.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/docs/supported-technologies.asciidoc)
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] Added an instrumentation plugin? Describe how you made sure that old, non-supported versions are not instrumented by accident.
- [x] This is something else
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)

### Test

I'll run https://github.com/elastic/apm-agent-java/actions/workflows/snapshot.yml after merging this PR.

### CODE REVIEW

**Review by JonasKunz — APPROVED** [review]

**Comment by v1v** [comment]
On hold for now. https://github.com/elastic/elastic-otel-java/pull/71 was merged, but there are some missing configurations when accessing the Vault secrets, so we are waiting for feedback from the system owners.

File: .buildkite/hooks/prepare-release.sh [hunk 1]
```diff
@@ -27,20 +12,22 @@ mkdir -p $GNUPGHOME
 chmod -R 700 $TMP_WORKSPACE
 
 echo "--- Prepare keys context :key:"
-VAULT_TOKEN=$(vault write -field=token auth/approle/login role_id="$VAULT_ROLE_ID_SECRET" secret_id="$VAULT_SECRET_ID_SECRET")
-export VAULT_TOKEN
-
 # Nexus credentials
-SERVER_USERNAME=$(vault read -field username secret/release/nexus)
+NEXUS_SECRET=kv/ci-shared/release-eng/team-release-secrets/apm/maven_central
+SERVER_USERNAME=$(vault read -field username $NEXUS_SECRET)
```
Comment (v1v) [inline .buildkite/hooks/prepare-release.sh:17]:
```suggestion
SERVER_USERNAME=$(vault kv get --field="username" $NEXUS_SECRET)
```

File: .buildkite/hooks/prepare-release.sh [hunk 2]
```diff
@@ -27,20 +12,22 @@ mkdir -p $GNUPGHOME
...
 export SERVER_USERNAME
-SERVER_PASSWORD=$(vault read -field password secret/release/nexus)
+SERVER_PASSWORD=$(vault read -field password $NEXUS_SECRET)
```
Comment (v1v) [inline .buildkite/hooks/prepare-release.sh:19]:
```suggestion
SERVER_PASSWORD=$(vault kv get --field="password" $NEXUS_SECRET)
```

File: .buildkite/hooks/prepare-release.sh [hunk 3]
```diff
@@ -27,20 +12,22 @@ mkdir -p $GNUPGHOME
...
 export SERVER_PASSWORD
 
 # Signing keys
-vault read -field=key secret/release/signing >$KEY_FILE
-KEYPASS_SECRET=$(vault read -field=passphrase secret/release/signing)
+GPG_SECRET=kv/data/ci-shared/release-eng/team-release-secrets/apm/gpg
+vault read -field="keyring" $GPG_SECRET | base64 -d > $KEY_FILE
```
Comment (v1v) [inline .buildkite/hooks/prepare-release.sh:24]:
```suggestion
vault kv get --field="keyring" $GPG_SECRET | base64 -d > $KEY_FILE
```

File: .buildkite/hooks/prepare-release.sh [hunk 4]
```diff
@@ -27,20 +12,22 @@ mkdir -p $GNUPGHOME
...
+## NOTE: passphase is the name of the field.
+KEYPASS_SECRET=$(vault read -field="passphase" $GPG_SECRET)
```
Comment (v1v) [inline .buildkite/hooks/prepare-release.sh:26]:
```suggestion
KEYPASS_SECRET=$(vault kv get --field="passphase" $GPG_SECRET)
```

File: .buildkite/hooks/prepare-release.sh [hunk 5]
```diff
@@ -27,20 +12,22 @@ mkdir -p $GNUPGHOME
...
+SERVER_USERNAME=$(vault kv get --field="username" $NEXUS_SECRET)
 export SERVER_USERNAME
-SERVER_PASSWORD=$(vault read -field password secret/release/nexus)
+SERVER_PASSWORD=$(vault kv get --field="password" $NEXUS_SECRET)
 export SERVER_PASSWORD
 
 # Signing keys
-vault read -field=key secret/release/signing >$KEY_FILE
-KEYPASS_SECRET=$(vault read -field=passphrase secret/release/signing)
+GPG_SECRET=kv/data/ci-shared/release-eng/team-release-secrets/apm/gpg
```
Comment (v1v) [inline .buildkite/hooks/prepare-release.sh:23]:
```suggestion
GPG_SECRET=kv/ci-shared/release-eng/team-release-secrets/apm/gpg
```

**Review by JonasKunz — APPROVED** [review]

**Pull request merged**